gno.land/cmd/gnoland/secrets.go

```diff
@@ -72,3 +76,84 @@ func constructSecretsPath(nodeDir string) string {
 		config.DefaultSecretsDir,
 	)
 }
+
+// printKeyValue searches and prints the given key value in JSON
+func printKeyValue[T *secrets | *config.Config](
+	input T,
+	raw bool,
+	io commands.IO,
+	key ...string,
+) error {
+	// prepareOutput prepares the JSON output, taking into account raw mode
+	prepareOutput := func(input any) (string, error) {
+		encoded, err := json.MarshalIndent(input, "", "    ")
+		if err != nil {
+			return "", fmt.Errorf("unable to marshal JSON, %w", err)
+		}
+
+		output := string(encoded)
+
+		if raw && len(encoded) > 0 && encoded[0] == '"' {
```

Comment:
It appears quite unstable.

Can we remove it and expect devops to use `RUN apk add jq` in their Dockerfile?

Alternatively, use `strconv.Unquote()`.

Reply:
`strconv.Unquote` should be sufficient, but it could be better to apply it only to string value and simply avoid printing the structure at all in raw mode.

Reply:
@moul for maintainability I would not suggest introducing such requirement from devops side.

Reply:
I wouldn't add `jq`, having `-r` is a simple use case which can work for a lot of script scenarios so I'd like to have it

I wouldn't use strconv.Unquote, I'd rather use

```go
var s string
err := json.Unmarshal(encoded, &s)
if err != nil {...}
return s, nil
```

I prefer this instead of checking if the type of a value is a string with reflect (ie. before marshaling to JSON); as this way the value of -r works correctly in case of types which use MarshalJSON or MarshalText.

Reply:
I've simplified it to @thehowl's suggestion:

[a11f035](https://github.com/gnolang/gno/pull/2393/commits/a11f0355144ad542a322c701c91dc7cf21cb7247)